src/CMakeLists.txt

```diff
@@ -59,6 +59,7 @@ set (sources
   World.cc
   ${PROTO_PRIVATE_SRC}
   ${network_sources}
+  systems/logical_audio_sensor_plugin/LogicalAudio.cc
```

Comment:
I think we should not add a system's source file to the core library since systems should be like plugins and only loaded when needed. Is this needed for the test? Maybe you can have a separate `ign_build_tests` call for the logical audio unit test?

Reply:
Just to clarify, `LogicalAudio.hh` and `LogicalAudio.cc` provide a set of audio functions that can be used by the logical audio plugin. The actual plugin is represented by `LogicalAudioSensorPlugin.hh` and `LogicalAudioSensorPlugin.cc`, and neither of the plugin files are being used here in `CMakeLists.txt`. So, in that case, are things okay here as the way they are?

Reply:
It's mainly because it's strange to see a system's file being part of the ign gazebo core library. If we plan to use this class in other parts of ign-gazebo we can consider moving the source files here in `src` but I think there is the plan of moving the logical audio sensor classes to ign-sensors in the future?

How about we move the logical audio sensor test cmake logic to ` systems/logical_audio_sensor_plugin/CMakeLists.txt`? e.g.


```
  
set (gtest_sources
  LogicalAudio_TEST.cc
) 
  
ign_build_tests(TYPE UNIT
  SOURCES
    ${gtest_sources}
  LIB_DEPS
    ${PROJECT_LIBRARY_TARGET_NAME}-logicalaudiosensorplugin-system
)
```

Reply:
That makes sense, thanks for the suggestion and CMake code. I went ahead and moved the test cmake logic to the plugin's CMakeLists.txt: https://github.com/ignitionrobotics/ign-gazebo/pull/401/commits/d52aa2ab4c9424cc989302721b473b6608c11b64